src/scripts/main.js

```diff
@@ -1,3 +1,357 @@
+  const nameLength = data.name.length;
+  const age = data.age;
+  const salary = data.salary;
+
+  return nameLength > 4 && (age >= 18 && age <= 90) && salary > 0;
+};
+
+const addNotification = type => {
+  let notification;
+
+  switch (type) {
+    case 'success':
+      notification = `
+        <div 
+          data-qa="notification" 
+          class="notification ${type}"
+        >
+          <p class="title">
+            The employee was successfully added. 
+          </p>
+        </div>
+      `;
+      break;
+    case 'error':
+      notification = `
+        <div 
+          data-qa="notification" 
+          class="notification ${type}"
+        >
+          <p class="title">
+            Invalid data.
+            Сheck the following:
+            * Name value has not less than 4 letters
+            * Age is not less than 18 or more than 90
+          </p>
+        </div>
+        `;
+      break;
+  }
+  root.insertAdjacentHTML('afterbegin', notification);
+
+  setTimeout(() => {
+    const message = root.querySelector('.notification');
+
+    message.remove();
+  }, 3000);
+};
+
+const makeSalaryValue = num => {
+  const correctNum = num.toFixed(3);
+
+  return `$${correctNum.toString().replace('.', ',')}`;
+};
+
+table.insertAdjacentHTML('afterend', formEl);
+
+const form = document.querySelector('form');
+
+form.addEventListener('submit', e => {
+  e.preventDefault();
+
+  const formData = new FormData(form);
+  const tableData = Object.fromEntries(formData.entries());
+
+  if (!formValidator(tableData)) {
+    addNotification('error');
+
+    return;
+  }
+
+  const newRow = `
+    <tr>
+      <td data-label="name">${tableData.name}</td>
+      <td data-label="position">${tableData.position}</td>
+      <td data-label="office">${tableData.office}</td>
+      <td data-label="age">${tableData.age}</td>
+      <td data-label="salary">${makeSalaryValue(+tableData.salary)}</td>
+    </tr>
+  `;
+
+  tBody.insertAdjacentHTML('beforeend', newRow);
+  addNotification('success');
+  form.reset();
+  setWidth();
+});
+
+// editing of table cells;
+
+const getCellType = (value) => {
+  let cellType;
+
+  switch (true) {
+    case value.includes('$'):
+      cellType = 'salary';
+      break;
+
+    case offices.includes(value):
+      cellType = 'office';
+      break;
+
+    case !isNaN(+value):
+      cellType = 'age';
+      break;
+
+    case typeof value === 'string':
+      cellType = 'str';
+      break;
+  }
+
+  return cellType;
+};
+
+const validateCell = (cellType, newValue) => {
+  switch (cellType) {
+    case 'str':
+      if (newValue.length > 2) {
+        return newValue;
+      }
+      break;
+
+    case 'office':
+      if (offices.includes(newValue)) {
+        return newValue;
+      }
+      break;
+
+    case 'age':
+      if (newValue >= 18 && newValue <= 90) {
+        return newValue;
+      }
+      break;
+
+    case 'salary':
+      if (!isNaN(parseFloat(newValue.replace('$', '')))) {
```

Comment:
https://github.com/mate-academy/js_employees_table_DOM/pull/223#issuecomment-1280406884
I don't think so
Initially, 183,000 means 183 thousand here:
![image](https://user-images.githubusercontent.com/86923089/198874051-896ecc5e-d19f-43c0-a069-1468afd31369.png)
Or $138,575 = 138 thousand and 575 dollars
So when you add a new row with 2000 salary it should be transformed to 2,000